homeassistant/components/meater/sensor.py

```diff
@@ -0,0 +1,188 @@
+    await coordinator.async_refresh()
+
+
+class MeaterProbeTemperature(Entity):
+    """Meater Temperature Sensor Entity."""
+
+    def __init__(self, coordinator, device_id, temperature_reading_type):
+        """Initialise the sensor."""
+        self.coordinator = coordinator
+        self.device_id = device_id
+        self.temperature_reading_type = temperature_reading_type
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return "Meater Probe " + self.temperature_reading_type.name
+
+    @property
+    def state(self):
+        """Return the temperature of the probe."""
+        # First find the right probe in the collection
+        device = None
+
+        for dev in self.coordinator.data:
+            if dev.id == self.device_id:
+                device = dev
+
+        if device is None:
+            return None
+
+        if TemperatureMeaturement.Internal == self.temperature_reading_type:
+            return device.internal_temperature
+
+        # Not an internal temperature, must be ambient
+        return device.ambient_temperature
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+
+        # Meater API always return temperature in Celsius
+        return TEMP_CELSIUS
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return "temperature"
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {
+                # Serial numbers are unique identifiers within a specific domain
+                (DOMAIN, self.device_id)
+            },
+            "name": self.name,
+            "manufacturer": "Apption Labs",
+        }
+
+    @property
+    def available(self):
+        """Return if entity is available."""
+        if not self.coordinator.last_update_success:
+            return False
+
+        # See if the device was returned from the API. If not, it's offline
+        device = None
+
+        for dev in self.coordinator.data:
+            if dev.id == self.device_id:
+                device = dev
+
+        return device is not None
+
+    @property
+    def should_poll(self):
+        """No need to poll. Coordinator notifies entity of updates."""
+        return False
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for the sensor."""
+        return f"{self.device_id}-{self.temperature_reading_type}"
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+    async def async_update(self):
+        """Update the entity.
+
+        Only used by the generic entity update service.
+        """
+        await self.coordinator.async_request_refresh()
+
+
+class TemperatureMeaturement(Enum):
```

Comment:
```suggestion
class TemperatureMeasurement(Enum):
```
Not sure if this was an intentional meat pun or not 😆.

Reply:
Few spelling errors I noticed

Reply:
Aha, unfortunately not an intentional pun. It's quite embarrassing, I thought I'd caught all the spelling mistakes. I'm mildly dyslexic, which makes programming hard; I once came back to a project two years after I'd completed it to do some upgrades and found a misspelled variable that I'd used throughout the codebase, just didn't spot it at the time.

Thanks again for catching those, I'm committing the changes now.

Reply:
No problem! I've been playing around with this and noticed one other small error in meater-python as it relates to the cook aspect of it. I know that doesn't impact this PR but it would if cook sensors were added at some point, so I threw in a PR on that repo. Hopefully that's alright. Thanks for your work on this!